applications/FluidDynamicsApplication/custom_elements/two_fluid_navier_stokes.cpp

```diff
@@ -157,7 +164,54 @@ void TwoFluidNavierStokes<TElementData>::CalculateLocalSystem(
                     ComputeGaussPointEnrichmentContributions(data, Vtot, Htot, Kee_tot, rhs_ee_tot);
                 }
 
-                CondenseEnrichment(data, rLeftHandSideMatrix, rRightHandSideVector, Htot, Vtot, Kee_tot, rhs_ee_tot);
+                if (rCurrentProcessInfo[SURFACE_TENSION]){
+                    // Surface tension coefficient is set in material properties
+                    const double surface_tension_coefficient = this->GetProperties().GetValue(SURFACE_TENSION_COEFFICIENT);
+                    Matrix int_shape_function, int_shape_function_enr_neg, int_shape_function_enr_pos;
+                    GeometryType::ShapeFunctionsGradientsType int_shape_derivatives;
+                    Vector int_gauss_pts_weights;
+                    std::vector<Vector> int_normals_neg;
+                    Vector gauss_pts_curvature;
+
+                    ComputeSplitInterface(
+                        data,
+                        int_shape_function,
+                        int_shape_function_enr_pos,
+                        int_shape_function_enr_neg,
+                        int_shape_derivatives,
+                        int_gauss_pts_weights,
+                        int_normals_neg,
+                        p_modified_sh_func);
+
+                    CalculateCurvature(
+                        int_shape_function,
+                        gauss_pts_curvature);
+
+                    SurfaceTension(
+                        surface_tension_coefficient,
+                        gauss_pts_curvature,
+                        int_gauss_pts_weights,
+                        int_shape_function,
+                        int_normals_neg,
+                        rRightHandSideVector);
+
+                    PressureGradientStabilization(
```

Comment:
Can this pressure gradient stabilization term be only applied in combination with the surface tension contribution?

Reply:
Generally, No. Therefor, although it is possible to mix them for the current code, I think it is better to keep a separate function for more clarity.

Reply:
Yes. But my question was if we could apply this same stabilization methodology for the standard two-fluids (with no surface tension). If so it is, I'd do it (or at least I'd make it optional).

Reply:
Theoretically, it is possible but I have not checked the required changes. Maybe, in the future I will do the modifications. So, please let it be unchanged for now.